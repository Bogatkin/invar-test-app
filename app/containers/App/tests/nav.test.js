import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import { Map, List } from 'immutable';
import Board from 'models/board';

import Nav from '../elements/nav';

const storeData = new Map({
  app: new Map({
    boards: new List([
      new Board({ id: 1, name: 'Test board 1' }),
      new Board({ id: 2, name: 'Test board 2' }),
    ]),
  }),
});
const mockStore = configureMockStore();

let store;

describe('<Nav />', () => {
  beforeEach(() => {
    store = mockStore(storeData);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Nav store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
