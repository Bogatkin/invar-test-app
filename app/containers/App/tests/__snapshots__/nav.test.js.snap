exports[`<Nav /> renders correctly 1`] = `
<nav
  className="nav has-shadow">
  <div
    className="container">
    <div
      className="nav-left">
      <a
        className="nav-item is-tab is-hidden-mobile is-active">
        Board 1
      </a>
      <a
        className="nav-item is-tab is-hidden-mobile">
        Board 2
      </a>
      <a
        className="nav-item is-tab is-hidden-mobile">
        Board 3
      </a>
    </div>
    <span
      className="nav-toggle">
      <span />
      <span />
      <span />
    </span>
  </div>
</nav>
`;
